driver_test.go

```diff
@@ -12,26 +12,291 @@
+		A, B, C string
+	}
+
+	tests := []struct {
+		name           string
+		input          string
+		want           []testQueryContextRow
+		wantErrorQuery bool
+		wantErrorScan  bool
+		wantErrorClose bool
+	}{
+		{
+			name:           "empty query",
+			wantErrorClose: true,
+			input:          "",
+			want:           []testQueryContextRow{},
+		},
+		{
+			name:           "syntax error",
+			wantErrorClose: true,
+			input:          "SELECT SELECT * FROM TestQueryContext",
+			want:           []testQueryContextRow{},
+		},
+		{
+			name:  "return nothing",
+			input: "SELECT * FROM TestQueryContext WHERE A = \"hihihi\"",
+			want:  []testQueryContextRow{},
+		},
+		{
+			name:  "select one tuple",
+			input: "SELECT * FROM TestQueryContext WHERE A = \"a1\"",
+			want: []testQueryContextRow{
+				{A: "a1", B: "b1", C: "c1"},
+			},
+		},
+		{
+			name:  "select subset of tuples",
+			input: "SELECT * FROM TestQueryContext WHERE A = \"a1\" OR A = \"a2\"",
+			want: []testQueryContextRow{
+				{A: "a1", B: "b1", C: "c1"},
+				{A: "a2", B: "b2", C: "c2"},
+			},
+		},
+		{
+			name:  "select subset of tuples with !=",
+			input: "SELECT * FROM TestQueryContext WHERE A != \"a3\"",
+			want: []testQueryContextRow{
+				{A: "a1", B: "b1", C: "c1"},
+				{A: "a2", B: "b2", C: "c2"},
+			},
+		},
+		{
+			name:  "select entire table",
+			input: "SELECT * FROM TestQueryContext ORDER BY A",
+			want: []testQueryContextRow{
+				{A: "a1", B: "b1", C: "c1"},
+				{A: "a2", B: "b2", C: "c2"},
+				{A: "a3", B: "b3", C: "c3"},
+			},
+		},
+		{
+			name:           "query non existant table",
+			wantErrorClose: true,
+			input:          "SELECT * FROM TestQueryContexta", want: []testQueryContextRow{},
+		},
+	}
+
+	// Run tests
+	for _, tc := range tests {
+
+		rows, err := db.QueryContext(ctx, tc.input)
+		if (err != nil) && (!tc.wantErrorQuery) {
+			t.Errorf("%s: unexpected query error: %v", tc.name, err)
+		}
+		if (err == nil) && (tc.wantErrorQuery) {
+			t.Errorf("%s: expected query error but error was %v", tc.name, err)
+		}
+
+		got := []testQueryContextRow{}
+		for rows.Next() {
+			var curr testQueryContextRow
+			err := rows.Scan(&curr.A, &curr.B, &curr.C)
+			if (err != nil) && (!tc.wantErrorScan) {
+				t.Errorf("%s: unexpected query error: %v", tc.name, err)
+			}
+			if (err == nil) && (tc.wantErrorScan) {
+				t.Errorf("%s: expected query error but error was %v", tc.name, err)
+			}
+
+			got = append(got, curr)
+		}
+
+		rows.Close()
+		err = rows.Err()
+		if (err != nil) && (!tc.wantErrorClose) {
+			t.Errorf("%s: unexpected query error: %v", tc.name, err)
+		}
+		if (err == nil) && (tc.wantErrorClose) {
+			t.Errorf("%s: expected query error but error was %v", tc.name, err)
+		}
+		if !reflect.DeepEqual(tc.want, got) {
+			t.Errorf("Test failed: %s. want: %v, got: %v", tc.name, tc.want, got)
+		}
+
+	}
+
+	// Drop table.
+	err = executeDdlApi(conn, []string{`DROP TABLE TestQueryContext`})
+	if err != nil {
+		t.Error(err)
+	}
+}
```

Comment:
nit: add new line add the end of the file.

Reply:
I tried adding it in but running go fmt before commit takes it out 🙁